docker/runit/init.sh

```diff
@@ -32,5 +32,19 @@ sed -i "s,<PROTOCOL>,${SPARK_SSL_PROTOCOL}," /etc/nginx/conf.d/spark.conf
 # bytes cipher strings of Java.
 # sed -i "s,<ENABLED_ALGORITHMS>,${SPARK_SSL_ENABLEDALGORITHMS//,/:}," /etc/nginx/conf.d/spark.conf
 
-# start service
-exec runsvdir -P /etc/service
+# Start runit processes (spark and nginx).
+exec runsvdir -P /etc/service &
+
+# The Spark process stdout is logged by runit's svlogd to a file named
+# "current".
+readonly SPARK_STDOUT_FILE="${MESOS_SANDBOX}"/spark/current
+
+# Wait for Spark process log file to exist.
+while [ ! -f "${SPARK_STDOUT_FILE}" ]; do
+  sleep 1
+done
+
+# Tail Spark process log file handled by runit. We do this because the output
+# from this script (which is run in the Marathon app's Docker container) is
+# logged by Mesos to the "stdout" files in the task sandbox.
+tail -F "${SPARK_STDOUT_FILE}"
```

Comment:
It looks like the main container process is now `tail` while service is running in the background. In case of a service failure `tail` will still be running. Wouldn't it be better to either run `tail` in the background or create a symlink for `$SPARK_STDOUT_FILE` like that: `ln -sf /dev/stdout $SPARK_STDOUT_FILE`

Reply:
+1. Either approach works and will allow you to get rid of the sleep (and potential infinite loop)

Reply:
I think running tail in the background works and is a better approach. The symlink approach doesn't work because the runit log files are rotated:
```
+ echo 'line 1'
+ cat current
line 1
+ ln -sf current stdout
+ cat current
line 1
+ cat stdout
line 1
+ echo 'line 2'
+ cat current
line 1
line 2
+ cat stdout
line 1
line 2
+ mv current current.1
+ echo 'line 3'
+ cat current
line 3
+ cat stdout
line 3
+ rm -rf current current.1
+ unlink stdout
```

Reply:
https://github.com/mesosphere/spark-build/pull/431/commits/9d0550a440d835e5ffce8f606639b052ba56251c